packages/client/src/App.test.tsx

```diff
@@ -1,9 +1,75 @@
-import React from "react"
-import ReactDOM from "react-dom"
+import { ListItem } from "@material-ui/core"
+import { createHistory, createMemorySource } from "@reach/router"
+import { ReactWrapper } from "enzyme"
+import * as React from "react"
 import App from "./App"
+import Avatar from "./Avatar"
+import { mount, updates } from "./testing"
+import * as $ from "./testing/fixtures"
 
-it("renders without crashing", () => {
-  const div = document.createElement("div")
-  ReactDOM.render(<App />, div)
-  ReactDOM.unmountComponentAtNode(div)
+it("displays a list of conversations", async () => {
+  const app = mount(<App />, {
+    mocks: [$.getAccountMock],
+    route: "accounts/1/dashboard"
+  })
+  await updates(app)
+  expect(app.find(ListItem)).toIncludeText("Hello from test")
 })
+
+it("archives a conversation from the list view", async () => {
+  const app = mount(<App />, {
+    mocks: [
+      $.getAccountMock,
+      $.archiveMock,
+      {
+        ...$.getAccountMock,
+        result: { data: { account: { ...$.account, conversations: [] } } }
+      }
+    ],
+    route: `accounts/${$.account.id}/dashboard`
+  })
+  await updates(app)
+  const row = findConversationRow(app, $.conversation)
+  expect(row).toIncludeText("Hello from test")
+  row.find(Avatar).simulate("click")
+  app.find('button[aria-label="archive"]').simulate("click")
+  await updates(app, 10)
+  expect(findConversationRow(app, $.conversation)).not.toExist()
+})
+
+it("archives a conversation from the conversation view", async () => {
+  const history = createHistory(
```

Comment:
Poodle uses @reach/router to render different React components when the URL changes. By default the router reads the location from `window.location`. But we can provide a different history source for the router to read and update location. In tests we use a "memory source" which stores the location in a plain javascript object. In this test I create a `history` object explicitly and pass it to `mount` so that I can check `history.location` after I expect the app to have changed the URL.